ocrd_cis/ocropy/binarize.py

```
@@ -105,146 +92,122 @@ def process(self):
-                        continue
-                    lines = region.get_TextLine()
-                    if not lines:
-                        self.logger.warning('Page "%s" region "%s" contains no text lines',
-                                            page_id, region.id)
-                    for line in lines:
-                        line_image, line_xywh = self.workspace.image_from_segment(
-                            line, region_image, region_xywh, feature_filter='binarized')
-                        self.process_line(line, line_image, line_xywh, zoom,
-                                          input_file.pageId, region.id,
-                                          file_id + '_' + region.id + '_' + line.id)
+        pcgts = input_pcgts[0]
+        page = pcgts.get_Page()
+        assert page
 
-            # update METS (add the PAGE file):
-            file_path = os.path.join(self.output_file_grp, file_id + '.xml')
-            pcgts.set_pcGtsId(file_id)
-            out = self.workspace.add_file(
-                ID=file_id,
-                file_grp=self.output_file_grp,
-                pageId=input_file.pageId,
-                local_filename=file_path,
-                mimetype=MIMETYPE_PAGE,
-                content=to_xml(pcgts))
-            self.logger.info('created file ID: %s, file_grp: %s, path: %s',
-                             file_id, self.output_file_grp, out.local_filename)
+        page_image, page_xywh, page_image_info = self.workspace.image_from_page(page, page_id, feature_filter='binarized')
+        zoom = determine_zoom(self.parameter['dpi'], page_image_info)
+        self.logger.info(f"Page '{page_id}' uses {self.parameter['dpi']} DPI")
+        
+        ret = [pcgts]
+        if level == 'page':
+            try:
+                ret.append(self.process_page(page, page_image, page_xywh, zoom, page_id, output_file_id))
+            except ValueError as e:
```

@bertsky Do we even want to catch these or should we let them explode and let core do the error handling? For page-wide binarization, I think we probably should because that means the page failed. But for region and line, we might have rogue instances of zero size but all the other regions/lines might be fine.

---

We will soon catch all exceptions on the page level in core. So this should not be handled here.

Regarding lower-level error handling: We have discussed this before, but partial failures across a page in general mean we also must be able to cope with partial annotation (incremental processors). We have no real solution ATM.

But since this PR just preserves the current behaviour (skipping partial failures regardless of level), and there are also other possible causes to catch in core, let's keep it like this.